src/realm/sync/client.cpp

```diff
@@ -1338,6 +1349,12 @@ MigrationStore* SessionWrapper::get_migration_store()
     return m_migration_store.get();
 }
 
+PendingResetStore* SessionWrapper::get_pending_reset_store()
+{
+    REALM_ASSERT(!m_finalized);
+    return m_pending_reset_store.get();
```

Comment:
You use `get_pending_reset_store()` in several places without any nullptr checks, could you add one here, or alternatively make this return a `PendingResetStore&` to communicate that it should never be null?

Reply:
I actually moved the PendingResetStore back to a set of `static` functions instead of an initialized object, since it's only used in a few places and to have control over the transactions and when they are committed.